### ISSUE TITLE
File Provider: max path length should be a capability

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#72768]
There is a questionable check in the explorer land for only allowing 255 characters for a file path:

https://github.com/Microsoft/vscode/blob/58b38d5ee54cbcc97cb864a49b16cab6439f00e3/src/vs/workbench/contrib/files/browser/fileActions.ts#L777

This should rather be a property the file system provider gives as information instead of being hard coded on Windows.

**Comment by isidorn** [comment]
This makes sense. Assigning to may to tackle then

**Comment by isidorn** [comment]
We first need to add the capability to the [`FileSystemProviderCapabilities`](https://github.com/Microsoft/vscode/blob/c14ce2ee7b2742276663faf8a9f165eacae05d02/src/vs/platform/files/common/files.ts#L195)
@bpasero would you like me to add this capability?
`FilePathLength`?
The issue I see is that the capabilities are a bit mask, and do we want this to make a boolean? Which sounds fishy, it should be a number which is set to 255 on windows.

Once we have the capability we can change the `validateFileName` to use the `fileService` to check for the capability.
Anyways, not for this week assigning to June.

**Comment by bpasero** [comment]
Maybe @jrieken has an idea. It seems fishy not being able to define the actual max length, but if we do not want to introduce something new then maybe the capability should be something like "max file path length 255 characters" so that by default this restriction does not apply.

**Comment by jrieken** [comment]
I though long path issues are a think of the past - doesn't windows allow you to workaround this? If not I would with Ben's suggestion and use a capability in that spirit of: names max length is 255 as I don't think there are other restrictions on other systems

**Comment by bpasero** [comment]
@jrieken yeah we could revisit this check, I think it was only there because the Windows Explorer would still not be able to handle such long paths. 

I believe there is still an absolute maximum of file name length (that is - just the name of the file, not its full path) that is true even on macOS and Linux.

**Comment by isidorn** [comment]
I just checked and the maximum file name on macOS is 256
On Windows10 I am allowed to create a full path length up to 256. So it is dependent on the path, on macOS it is not.
Tried copying over a long file name from macOS to windows, but windows could not handle it. So there still seems to be the 256 restriction.

**Comment by bpasero** [comment]
@isidorn you can also just use node.js from the command line to create files > 255path length for testing. node.js can handle it, but the Windows Explorer I guess cannot.

I think the filename limit is consistent across all platforms and probably is fine to be hardcoded.

**Comment by isidorn** [comment]
I just tried creating a file with long path from the windows terminal it also does not allow me (complains about full path length). However I could create it using a node program. That long file path does not get displayed in the windows explorer but gets nicely displayed in the terminal.

Due to that I decided to remove the check.

Note that there is still the check for the filename being longer than 255 [here](https://github.com/Microsoft/vscode/blob/ec1e0c657e7c5129d29d32060c2c027e43eb962c/src/vs/base/common/extpath.ts#L174) which is used from this method for every platform.
